File: image_generation.py
import os
from openai import OpenAI
from dotenv import load_dotenv

# Load environment variables from .env file
load_dotenv()

# Get API key from environment variable
api_key = os.getenv("OPENAI_API_KEY")

# Initialize OpenAI client
client = OpenAI(api_key)

def generate_image(prompt, size="1024x1024", quality="standard", n=1):
    response = client.images.generate(
        model="dall-e-3",
        prompt=prompt,
        size=size,
        quality=quality,
        n=n,
    )

    return response.data[0].url

if __name__ == "__main__":
    prompt = "a white siamese cat"
    image_url = generate_image(prompt)
    print(f"Generated image URL: {image_url}")

File: image_generation.js
const OpenAI = require('openai');
const fs = require('fs');
require('dotenv').config();

const openai = new OpenAI(process.env.OPENAI_API_KEY);

async function generateImage() {
    const prompt = "a white siamese cat";
    const size = "1024x1024";
    const quality = "standard";
    const n = 1;

    try {
        const response = await openai.images.generate({
            model: "dall-e-3",
            prompt: prompt,
            size: size,
            quality: quality,
            n: n
        });

        const imageUrl = response.data[0].url;
        console.log(imageUrl);
    } catch (error) {
        if (error.response) {
            console.log(error.response.status);
            console.log(error.response.data);
        } else {
            console.log(error.message);
        }
    }
}

generateImage();
